docs/design/2023-04-11-pause-user-ddl-when-upgrading.md

```diff
@@ -0,0 +1,112 @@
+# Proposal: Pause User DDL when Upgrading TiDB
+
+* Author: [zimulala](https://github.com/zimulala)
+* Tracking issue: https://github.com/pingcap/tidb/issues/39751
+
+## Abstract
+
+This document describes a feature that allows users to pause the execution of their Data Definition Language (DDL) statements during the upgrade process of TiDB. During the upgrade process, the system stops the execution of user DDL statements, completes the system DDL operation, and finally resumes the execution of the original user DDL statements. This feature is available for upgrades from v7.1 and later versions of TiDB, and it does not support upgrades from versions prior to v7.1.
+
+## Motivation or Background
+
+Since the cluster upgrade may need to deal with DDL statements(there may be some DDL statements being processed before the upgrade), and DDL itself implementation framework will also need to adjust, resulting in some versions may not be able to roll the upgrade situation. Although these cases only exist in some versions, it is not easy to describe them one by one. Therefore, when upgrading TiDB clusters, users need to confirm that there are no DDL statements being executed, otherwise there may be problems with undefined behavior.
```

Comment:
```suggestion
1. Since the TiDB cluster upgrade may need to deal with DDL statements(there may be some DDL statements being processed before the upgrade).
2. DDL module itself implementation logic will also need to be adjusted, resulting in some versions may not allow user DDL job running during roll upgrade period. 
Although above scenarios may only exist in some upgrade cases, it is not easy to describe them one by one. Therefore, when upgrading TiDB clusters, users need to first confirm that there are no DDL statements being executed, otherwise there may be problems with unexpection behavior.
```